lib/assets.js

```diff
@@ -102,17 +110,27 @@ Assets.prototype.serveAsset = function (req, res, next) {
 
   var contentType = asset.contentType;
 
-  if (contentType.match(/^text\/|\/json$|\/javascript$/)) {
-    contentType += "; charset=UTF-8";
-  }
+  if (contentType) {
+    if (contentType.match(/^text\/|\/json$|\/javascript$/)) {
+      contentType += "; charset=UTF-8";
+    }
 
-  res.setHeader("Content-Type", contentType);
+    res.setHeader("Content-Type", contentType);
+  }
   res.setHeader("Content-Length", asset.length);
 
   if (req.method === "HEAD") {
     return res.end();
   }
 
+  if (!asset.buffer) {
+    var localFilePath = pathModule.join(this.options.buildDir, assetName);
+    return fs.readFile(localFilePath, function (err, data) {
+      if (err) return next(err);
+      res.end(data.toString());
```

Comment:
We probably want to avoid calling `toString()` here, since it unnecessarily converts our raw Buffer into a V8 String, and then back into a raw Buffer.